### ISSUE TITLE
Annotation inheritance from interfaces broken

### ISSUE THREAD

**Issue by eradford** [swagger-api/swagger-core#2644]
| Q                                 | A
| --------------------------------- | -------
| Bug or feature request?           | Bug|
| Which Swagger-Core version?       |2.0|
| Which Java version?               |8|
| Which JAX-RS framework & version? |Apache CXF 3.1.6|
 

The class scanning logic in swagger-jaxrs2 does not properly process annotations which should be inherited from an interface.

The conditional here is the root cause of the issue:
https://github.com/swagger-api/swagger-core/blob/b1fd067e65ea1a2bd8e0d8317b8fdfee6447948a/modules/swagger-core/src/main/java/io/swagger/v3/core/util/ReflectionUtils.java#L305

Since `annotations` is ultimately defined a few lines above as:
https://github.com/swagger-api/swagger-core/blob/b1fd067e65ea1a2bd8e0d8317b8fdfee6447948a/modules/swagger-core/src/main/java/io/swagger/v3/core/util/ReflectionUtils.java#L292

and [Class#getAnnotationsByType(Class)](https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getAnnotationsByType-java.lang.Class-) defines it's return value to be:

> Returns: all this element's annotations for the specified annotation type if associated with this element, else an array of length zero

the above condition will never be true, and implemented interfaces will never be scanned for annotations, even if no classes in the hierarchy contain any Swagger or jax-rs annotations.

The basic flow of the problem method is:
```
annotations = get annotations for the provided class
if annotations is null or empty
    annotations = get annotations for the provided class's super class
if annotations is still **strictly** null
    annotations = get annotations from the interfaces that the provided class implements
if annotations is still **strictly** null
    return null
return annotations
```

It appears as though it was a mere oversight that the second and third conditional do not check for an empty array in addition to null.

To sum things up:

Given this interface:

```
@Path("resources")
@Tag(name = "resource")
public interface AnnotatedInterface {
    @GET
    Response getResource();
}
```

```
public class ConcreteImplementation implements AnnotatedInterface {
    public Response getResource() {
        // Get the resource
    }
}
```

Observed behavior: `GET /resources` appears only under the default tag, and nowhere else
Expected behavior: `GET /resources` appears under the 'resource' tag

**Comment by eradford** [comment]
I've already crafted a test that stresses the bad logic and implemented the code fix. I will be submitting a pull request for this shortly.

Just wanted to create this issue to document the problem and track work on it.

**Comment by frantuma** [comment]
Thanks for reporting and for the fix! Included in merged #2650, closing ticket, please reopen if you're still experiencing issues
